Application/EdFi.Ods.Common/Validation/Validator.cs

```diff
@@ -0,0 +1,692 @@
+        // [RequiresUnreferencedCode(ValidationContext.InstanceTypeNotStaticallyDiscovered)]
+        private static List<ValidationError> GetObjectPropertyValidationErrors(object instance,
+            ValidationContext validationContext, bool validateAllProperties, bool breakOnFirstError)
+        {
+            var properties = GetPropertyValues(instance, validationContext);
+            var errors = new List<ValidationError>();
+
+            foreach (var property in properties)
+            {
+                // get list of all validation attributes for this property
+                var attributes = _store.GetPropertyValidationAttributes(property.Key);
+
+                if (validateAllProperties)
+                {
+                    // validate all validation attributes on this property
+                    errors.AddRange(GetValidationErrors(property.Value, property.Key, attributes, breakOnFirstError));
+                }
+                else
+                {
+                    // only validate the first Required attribute
+                    foreach (ValidationAttribute attribute in attributes)
+                    {
+                        if (attribute is RequiredAttribute reqAttr)
+                        {
+                            // Note: we let the [Required] attribute do its own null testing,
+                            // since the user may have subclassed it and have a deeper meaning to what 'required' means
+                            var validationResult = reqAttr.GetValidationResult(property.Value, property.Key);
+                            if (validationResult != ValidationResult.Success)
+                            {
+                                errors.Add(new ValidationError(reqAttr, property.Value, validationResult!));
+                            }
+                            break;
+                        }
+                    }
+                }
+
+                if (breakOnFirstError && errors.Count > 0)
+                {
+                    break;
+                }
+            }
+
+            return errors;
+        }
+
+        /// <summary>
+        ///     Retrieves the property values for the given instance.
+        /// </summary>
+        /// <param name="instance">Instance from which to fetch the properties.</param>
+        /// <param name="validationContext">Describes the entity being validated.</param>
+        /// <returns>
+        ///     A set of key value pairs, where the key is a validation context for the property and the value is its current
+        ///     value.
+        /// </returns>
+        /// <remarks>Ignores indexed properties.</remarks>
+        // [RequiresUnreferencedCode(ValidationContext.InstanceTypeNotStaticallyDiscovered)]
+        private static List<KeyValuePair<ValidationContext, object?>> GetPropertyValues(object instance,
+            ValidationContext validationContext)
+        {
+            // ==========================
```

Comment:
All customizations made to the code is commented with the text: `// NOTE: Customization`